Set serviceClass on WebRTC network connections

youennf opened the pull request:
#### 3bf07acdb9cce82d79e51c19e940fd4cc8e7b3d9
<pre>
Set serviceClass on WebRTC network connections
<a href="https://rdar.apple.com/149874300">rdar://149874300</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=291641">https://bugs.webkit.org/show_bug.cgi?id=291641</a>

Reviewed by Jean-Yves Avenard.

We introduce a feature flag to ocntrol whether marking the service class of a given nw connection (used for WebRTC) as video.
This can help the OS to optimize things for that nw connection.
The flag is off by default for now.
The patch is mostly plumbing the flag from the web process up to nw connections living in networking process.
We do a refactoring to pass the 3 booleans used for creating nw sockets as a structure called RTCSocketCreationFlags.
This makes the code clearer and less error prone.

Canonical link: <a href="https://commits.webkit.org/301152@main">https://commits.webkit.org/301152@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bdf4ead7a3866f60ecf5f6419b7312091895e413

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125074 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44742 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35480 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131922 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76949 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5dce7989-1ac0-4a5d-9e20-58ab46344d0c) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45433 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53307 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95208 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63179 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/4bc3c143-503b-474a-9c6b-b2581ab744f0) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128028 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36267 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111862 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75750 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b64f525a-d2ee-4866-aebd-666160a836e9) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35179 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30017 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75403 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/117169 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106031 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30246 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134600 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/123592 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51892 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39686 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103677 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52319 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108074 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103449 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26344 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48798 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27091 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48945 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51779 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57571 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/156615 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51151 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39226 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54507 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52842 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
```diff
@@ -255,7 +255,7 @@ const String& NetworkRTCProvider::attributedBundleIdentifierFromPageIdentifier(W
 }
 
-void NetworkRTCProvider::createUDPSocket(LibWebRTCSocketIdentifier identifier, const RTCNetwork::SocketAddress& address, uint16_t minPort, uint16_t maxPort, WebPageProxyIdentifier pageIdentifier, bool isFirstParty, bool isRelayDisabled, WebCore::RegistrableDomain&& domain)
+void NetworkRTCProvider::createUDPSocket(LibWebRTCSocketIdentifier identifier, const RTCNetwork::SocketAddress& address, uint16_t minPort, uint16_t maxPort, WebPageProxyIdentifier pageIdentifier, bool isFirstParty, bool isRelayDisabled, bool enableServiceClass, WebCore::RegistrableDomain&& domain)
```
can we reduce this to a structure instead? there's too many bool to follow

youennf replied:
Done, this is indeed better

jyavenard commented on Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.messages.in:
```diff
@@ -29,14 +29,14 @@
 
 #if PLATFORM(COCOA)
-    GetInterfaceName(URL url, WebKit::WebPageProxyIdentifier pageIdentifier, bool isFirstParty, bool isRelayDisabled, WebCore::RegistrableDomain domain) -> (String interfaceName) async
+    GetInterfaceName(URL url, WebKit::WebPageProxyIdentifier pageIdentifier, bool isFirstParty, bool isRelayDisabled, bool enableServiceClass, WebCore::RegistrableDomain domain) -> (String interfaceName) async
```
same here. too many bools

jyavenard reviewed (approved):
r=me with nits. This code is starting to be quite messy with lots of arguments of similar type ; it's extremely easy to pass them wrong and get unpredictable results.

Should pass a structure instead.

youennf committed:
Set serviceClass on WebRTC network connections
rdar://149874300
https://bugs.webkit.org/show_bug.cgi?id=291641

Reviewed by Jean-Yves Avenard.

We introduce a feature flag to ocntrol whether marking the service class of a given nw connection (used for WebRTC) as video.
This can help the OS to optimize things for that nw connection.
The flag is off by default for now.
The patch is mostly plumbing the flag from the web process up to nw connections living in networking process.
We do a refactoring to pass the 3 booleans used for creating nw sockets as a structure called RTCSocketCreationFlags.
This makes the code clearer and less error prone.

Canonical link: https://commits.webkit.org/301152@main
